pkg/controllers/service/service_lb_controller.go

```diff
@@ -0,0 +1,155 @@
+	_ "github.com/vmware-tanzu/nsx-operator/pkg/nsx/ratelimiter"
+	servicecommon "github.com/vmware-tanzu/nsx-operator/pkg/nsx/services/common"
+)
+
+var (
+	log           = logger.Log
+	ResultNormal  = common.ResultNormal
+	ResultRequeue = common.ResultRequeue
+	MetricResType = common.MetricResTypeServiceLb
+)
+
+// ServiceLbReconciler reconciles a Service LoadBalancer object
+type ServiceLbReconciler struct {
+	Client   client.Client
+	Scheme   *apimachineryruntime.Scheme
+	Service  *servicecommon.Service
+	Recorder record.EventRecorder
+}
+
+func updateSuccess(r *ServiceLbReconciler, c *context.Context, lbService *v1.Service) {
+	r.setServiceLbStatus(c, lbService)
+	r.Recorder.Event(lbService, v1.EventTypeNormal, common.ReasonSuccessfulUpdate, "LoadBalancer service has been successfully updated")
+	metrics.CounterInc(r.Service.NSXConfig, metrics.ControllerUpdateSuccessTotal, MetricResType)
+}
+
+func (r *ServiceLbReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	service := &v1.Service{}
+
+	if err := r.Client.Get(ctx, req.NamespacedName, service); err != nil {
+		log.Error(err, "unable to fetch lb service", "req", req.NamespacedName)
+		return ResultNormal, client.IgnoreNotFound(err)
+	}
+
+	if service.Spec.Type == v1.ServiceTypeLoadBalancer {
+		log.Info("reconciling lb service", "lbService", req.NamespacedName)
+		metrics.CounterInc(r.Service.NSXConfig, metrics.ControllerSyncTotal, MetricResType)
+
+		if service.ObjectMeta.DeletionTimestamp.IsZero() && r.Service.NSXConfig.EnableVPCNetwork {
+			metrics.CounterInc(r.Service.NSXConfig, metrics.ControllerUpdateTotal, MetricResType)
+			updateSuccess(r, &ctx, service)
+		}
+	}
+
+	return ResultNormal, nil
+}
+
+func (r *ServiceLbReconciler) setServiceLbStatus(ctx *context.Context, lbService *v1.Service) {
+	ipMode := v1.LoadBalancerIPModeProxy
+	statusUpdated := false
+	// If tanzu.vmware.com/ingress-ip-mode label with values proxy or vip,
+	// the LoadBalancer serivice ipMode status would be set to whatever the label is set to,
+	// Otherwise, it's set to Proxy by default when unset or other invalid values.
+	if labelIpMode, ok := lbService.Labels[servicecommon.LabelLbIngressIpMode]; ok {
+		if labelIpMode == servicecommon.LabelLbIngressIpModeVipValue {
+			ipMode = v1.LoadBalancerIPModeVIP
+		}
+	}
+	for i, ing := range lbService.Status.LoadBalancer.Ingress {
+		if ing.IPMode == nil || *(ing.IPMode) != ipMode {
+			lbService.Status.LoadBalancer.Ingress[i].IPMode = &ipMode
+			statusUpdated = true
+		}
+	}
+
+	if statusUpdated {
+		r.Client.Status().Update(*ctx, lbService)
+		log.V(1).Info("updated LB service status ipMode", "Name", lbService.Name, "Namespace", lbService.Namespace, "ipMode", ipMode)
+	}
+}
+
+func (r *ServiceLbReconciler) setupWithManager(mgr ctrl.Manager) error {
+	return ctrl.NewControllerManagedBy(mgr).
+		For(&v1.Service{}).
+		WithOptions(
+			controller.Options{
+				MaxConcurrentReconciles: runtime.NumCPU(),
```

Comment:
runtime.NumCPU() -> common.NumReconcile()

Reply:
Yes, correct. we should change it to  common.NumReconcile(). Thanks for advice.